File: src/main/java/net/onelitefeather/attollo/listener/UpdateCheckerListener.java
package net.onelitefeather.attollo.listener;

import net.onelitefeather.attollo.service.UpdateService;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public final class UpdateCheckerListener implements Listener {

    private final UpdateService updateService;

    public UpdateCheckerListener(UpdateService updateService) {
        this.updateService = updateService;
    }

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        var player = event.getPlayer();
        if (player.isOp() || player.hasPermission(this.updateService.getPermission())) {
            this.updateService.notifyAudience(event.getPlayer());
        }
    }

}

File: src/main/kotlin/dev/themeinerlp/attollo/listener/UpdateCheckerListener.kt
package dev.themeinerlp.attollo.listener

import dev.themeinerlp.attollo.Attollo
import dev.themeinerlp.attollo.NOTIFY_UPDATE_PERMISSION
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class UpdateCheckerListener(private val attollo: Attollo) : Listener {

    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        val player = event.player
        if (player.isOp || player.hasPermission(NOTIFY_UPDATE_PERMISSION)) {
            attollo.updateService.notifyPlayer(player)
        }
    }

}